src/js/components/Menu/Menu.js

```diff
@@ -332,11 +330,7 @@ const Menu = forwardRef((props, ref) => {
               {alignControlMirror === 'top' && align.top === 'top'
                 ? controlMirror
                 : undefined}
-              <Box
-                overflow="auto"
-                role="menu"
-                a11yTitle={ariaLabel || a11yTitle}
-              >
+              <Box overflow="auto" role="menu" a11yTitle={dropA11y || a11y}>
```

Comment:
We are already passing `dropProps` to the DropButton at line 312. I think anything passed via `dropProps` should be handled by Drop itself. In this particular case, I think we should be changing DropContainer.js to set the `aria-label` on StyledDrop around line 331.